src/viser/client/src/CameraControls.tsx

```diff
@@ -113,11 +113,17 @@ export function SynchronizedCameraControls() {
     setTimeout(() => sendCamera(), 50);
   }, [connected, sendCamera]);
 
+  // Send camera for 3D viewport changes.
   React.useEffect(() => {
-    window.addEventListener("resize", sendCamera);
-    return () => {
-      window.removeEventListener("resize", sendCamera);
-    };
+    // Create a resize observer to resize the CSS canvas when the window is resized.
+    const resizeObserver = new ResizeObserver(() => { sendCamera() });
+
+    // Observe the canvas.
+    const canvas = viewer.canvasRef.current!  // r3f viewer canvas
+    resizeObserver.observe(canvas);
+
+    // Cleanup
```

Comment:
can we add some punctuation :D

Reply:
I think this is addressed in [239bf63](https://github.com/nerfstudio-project/viser/pull/178/commits/239bf631db210bf6825610658e50d932b9a86fbf), let me know if this was not what you meant.